multidict/multidict_api.h

```diff
@@ -22,60 +23,348 @@ typedef struct {
        if the client is compiled with older multidict_api.h header
     */
 
-    void *state;
+    void* state;
 
-    PyTypeObject *(*MultiDict_GetType)(void *state);
+    PyTypeObject* (*MultiDict_GetType)(void* state);
+
+    PyObject* (*MultiDict_New)(void* state, int prealloc_size);
+    int (*MultiDict_Add)(void* state, PyObject* self, PyObject* key,
+                         PyObject* value);
+    int (*MultiDict_Clear)(void* state, PyObject* self);
+
+    PyObject* (*MultiDict_SetDefault)(void* state, PyObject* self,
+                                      PyObject* key, PyObject* _default);
+
+    int (*MultiDict_Del)(void* state, PyObject* self, PyObject* key);
+    uint64_t (*MultiDict_Version)(void* state, PyObject* self);
+
+    // proposed but IDK yet...
+    // static int MultiDict_CreatePosMarker(void* state, PyObject* self,
+    // md_pos_t* pos)
+
+    // static int MultiDict_Next(void* state, PyObject* self,
```

Comment:
I'm still not sure what API should be; let's implement the function later to simplify the review